link and information extractor

shaibrown5 opened the pull request:
this script executes 3 tasks:
1. webscrape the url's from the list and get the zip file links
2. extract the xml document in each zip file,
3. extract the relevant info from each xml tree and place in the relevant db table

the script for placing in the table is written to accommodate 1 link from each super. this will be cahnges

shaibrown5 committed:
This script executes 3 tasks:
1. webscrape the url's from the given url list and find the zip file
link
2. extract the xml file from the zip and parse it
3. extract the relevant info from the xml tree and place into the
relevant database tables

very slow as it has to go through almost all the links,
sql table input is an example only used for 2 links

shaibrown5 committed:
Update information_extractor.py

shaibrown5 commented on information_extractor.py:
```diff
@@ -0,0 +1,93 @@
+        # gets relevant child
+        items = tree.getchildren()[-1]
+        extract_information(items)
+        break  # only use 1 link from each url
```
this was left behind from testing, i only wanted 1 link from each url

simzacks commented on information_extractor.py:
```diff
@@ -0,0 +1,93 @@
+
+        item_info_formula = """INSERT INTO item_info (itemCode,itemId,itemName, itemPrice) VALUES(%s,%s,%s,%s) ON DUPLICATE KEY UPDATE itemPrice = %s"""
+        price_table_shufersal = """INSERT INTO price_table (itemCode, shufersal) VALUES(%s, %s)  ON DUPLICATE KEY UPDATE shufersal = %s """
+        price_table_mega = """INSERT INTO price_table (itemCode, mega) VALUES(%s, %s)  ON DUPLICATE KEY UPDATE mega = %s"""
```
A more scalable price_table would have :
itemcode, store_branch_id, price
the price may be different per branch, and you don't want to have to add every branch you are going to be using as a new column. 
Also item_info shouldn't have price, because that's different per branch.

shaibrown5 replied:
so to see that i understand; a price_table for each super market, in which the primary keys are the item code and the branch id. Should keep a single item_info table? if so what do you recommend i do in case of duplicate keys with different  product names (as in both are 3% milk, but one is named "milk" the other is "3% milk"?

shaibrown5 committed:
Merge branch 'master' of https://github.com/beyond-io/superx

shaibrown5 committed:
removed item price from info table

shaibrown5 committed:
changes zip_fils into a set, added .decode method

after some debugging saw that multiples if not more of the saem link were entered into the zip_links , and so I changed zip_links into a set

shaibrown5 committed:
added a third super market and made relevant changes to accomidate it

shaibrown5 committed:
Merge branch 'master' of https://github.com/shaibrown5/superx

shaibrown5 committed:
Merge branch 'master' of https://github.com/beyond-io/superx

shaibrown5 commented:
accidentaly deleted file, i will make another pr
